src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/RazorProjectInfoFactory.cs

```diff
@@ -35,6 +35,11 @@ static RazorProjectInfoFactory()
 
     public static async Task<RazorProjectInfo?> ConvertAsync(Project project, CancellationToken cancellationToken)
     {
+        if (project.FilePath is null)
+        {
+            return null;
+        }
+
         var projectPath = Path.GetDirectoryName(project.FilePath);
         if (projectPath is null)
```

Comment:
is it possible for this to be null if the `project.FilePath` is checked above?

Reply:
I don't believe so, but it's annotated to be `string?` on .NET without a `[NotNullIfNotNull]` attribute.

Reply:
Actually, it turns out that it *is* possible for this to return null if `project.FilePath` is not null. `Project.GetDirectoryName(...)` returns null if the file path pass to it is null OR if it is a root without a directory name, such as `/` or `C:`.